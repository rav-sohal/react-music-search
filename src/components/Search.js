import React, {Component} from 'react';
import {Jumbotron} from 'react-bootstrap';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const style = {
  textAlign: 'center',
};

const SearchDiv = styled.div`
    background: lightGrey;
    padding: 35px;
`;

const SearchInput = styled.input`
    height: 48px;
`;

class Search extends Component {
  state = {
    searchText: '',
  };

  searchInput = React.createRef ();

  onSearchChange = e => {
    this.setState ({searchText: e.target.value});
  };

  handleSubmit = e => {
    e.preventDefault ();
    this.props.onSearch (this.searchQuery.value);
    e.currentTarget.reset ();
  };

  render (props) {
    return (
      <Jumbotron style={style}>
        <div className="container">
          <h1>Search iTunes Music</h1>
          <p>
            Search your favourite artist, album or song.
          </p>
          <SearchDiv>
            <form className="form-group" onSubmit={this.handleSubmit}>
              <div className="row">
                <div className="col-md-9">
                  <SearchInput
                    type="search"
                    onChange={this.onSearchChange}
                    name="search"
                    ref={input => (this.searchQuery = input)}
                    placeholder="Search..."
                    className="form-control"
                  />
                </div>
                <div className="col-md-3">
                  <button
                    type="submit"
                    id="submit"
                    className="btn btn-success btn-lg btn-block sm-mt-3"
                  >
                    Search
                  </button>
                </div>
              </div>
            </form>
          </SearchDiv>
        </div>
      </Jumbotron>
    );
  }
}

Search.propTypes = {
  onSearch: PropTypes.func,
};

export default Search;
