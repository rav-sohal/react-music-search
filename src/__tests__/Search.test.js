import {render, screen} from '@testing-library/react';
import {create} from 'react-test-renderer';
import Search from '../components/Search';

describe ('Search.js tests', () => {
  
  test ('Search button class', () => {
    render (<Search />);
    const button = document.getElementById ('submit');
    expect (button).toHaveAttribute ('type', 'submit');
  });

  test ('Snapshot test of Search', () => {
    const testSearchSnapshot = create (<Search />).toJSON ();
    expect (testSearchSnapshot).toMatchSnapshot ();
  });
  
});
