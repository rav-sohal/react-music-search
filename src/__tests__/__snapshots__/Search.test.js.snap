// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search.js tests Snapshot test of Search 1`] = `
<div
  className="jumbotron"
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <div
    className="container"
  >
    <h1>
      Search iTunes Music
    </h1>
    <p>
      Search your favourite artist, album or song.
    </p>
    <div
      className="sc-bdvvaa gYYXMo"
    >
      <form
        className="form-group"
        onSubmit={[Function]}
      >
        <div
          className="row"
        >
          <div
            className="col-md-9"
          >
            <input
              className="sc-gsDJrp hzTNRY form-control"
              name="search"
              onChange={[Function]}
              placeholder="Search..."
              type="search"
            />
          </div>
          <div
            className="col-md-3"
          >
            <button
              className="btn btn-success btn-lg btn-block sm-mt-3"
              id="submit"
              type="submit"
            >
              Search
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
